lib/model/model.go

```diff
@@ -2882,16 +2882,30 @@ func (m *model) Availability(folder string, file protocol.FileInfo, block protoc
 	}
 	defer snap.Release()
 
-	return m.availabilityInSnapshotRLocked(cfg, snap, file, block), nil
+	if devices := m.availabilityInSnapshot(cfg, snap, file); len(devices) > 0 {
+		return devices, nil
```

Comment:
Previous behaviour wasn't to abort/return here, but to continue with temporary downloads and append any devices found there.